src/cyrsasl_external.erl

```diff
@@ -0,0 +1,61 @@
+-module(cyrsasl_external).
+-author('denys.gonchar@erlang-solutions.com').
+
+-xep([{xep, 178}, {version, "1.1"}, {comment, "partially implemented."}]).
+
+-export([start/1, stop/0, mech_new/2, mech_step/2]).
+
+-behaviour(cyrsasl).
+
+-record(state, {creds}).
```

Comment:
saslext_state would be a better name.
Because sometimes we do load records definitions into shell, and it works better, if each record have a unique name.

But maybe consider to use maps.
creds should be typed and commented.

Reply:
We use `state` for record name in so many places, that I think it should be a coordinated refactoring effort (if we would like to do it at all).

Reply:
But :+1: for typed `creds`.